Core/NPCBarrierGlow.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.Graphics.Effects;
+using System;
+using Terraria;
+using Terraria.ModLoader;
+using StarlightRiver.Content.CustomHooks;
+
+namespace StarlightRiver.Core
+{ 
+    public class NPCBarrierGlowSystem : ModSystem
+    {
+        public override void PreUpdateNPCs()
+        {
+            NPCBarrierGlow.anyEnemiesWithBarrier = false;
+        }
+    }
+    public class NPCBarrierGlow : IOrderedLoadable
```

Comment:
Whats the purpose of this being a seperate class apart from the ModSystem? ModSystem has its own load hooks